src/ziggurat/init.clj

```diff
@@ -88,18 +93,36 @@
   (mount/stop #'server/server)
   (stop-messaging))
 
+(defn start-batch-consumer [args]
+  (-> (mount/only #{#'thread-pool/executor-thread-pool
+                    #'consumer-driver/consumer-groups})
+      (mount/with-args (:batch-routes args))
+      (mount/start)))
+
+(defn stop-batch-consumer []
+  (-> (mount/only #{#'thread-pool/executor-thread-pool
+                    #'consumer-driver/consumer-groups})
+      (mount/stop)))
+
 (def valid-modes-fns
   {:api-server     {:start-fn start-server :stop-fn stop-server}
    :stream-worker  {:start-fn start-stream :stop-fn stop-stream}
    :worker         {:start-fn start-workers :stop-fn stop-workers}
+   :batch-consumer {:start-fn start-batch-consumer :stop-fn stop-batch-consumer}
    :management-api {:start-fn start-management-apis :stop-fn stop-management-apis}})
 
+(defn- all-modes []
+  (keys valid-modes-fns))
+
+(defn- all-modes-except-batch-consumer []
```

Comment:
Why do we need this?

Reply:
**Short Answer:**
Thus, we need to exclude this to maintain backward compatibility.

**Long Answer:**
When an arity of main is 4 (old style), only  stream-routes are supported. At the same time, validations fail if  one does not provide the routes for the given  mode.

Now, because mode is nil (in arity of 4) which means all the modes are activated by default, batch-worker mode gets activated too and the validations fail.